src/main/java/com/github/qtrouper/Trouper.java

```diff
@@ -89,6 +91,12 @@ private boolean handle(Message message, AMQP.BasicProperties properties) throws
 
         if (processed) return true;
 
+        boolean expiresAtEnabled = (boolean) properties.getHeaders().getOrDefault(EXPIRES_AT_ENABLED, false);
```

Comment:
1. Cast into Container classes only (Boolean) (Long) etc, instead of boolean, long. 
2. Move this check before the process method is called. Otherwise this might not work.

Reply:
Missed 2nd while copying line by line to forked repo as style sheet was not the same as yours. Updated now.